content/modules/rest/pages/entities-api.adoc

```diff
@@ -0,0 +1,724 @@
+.Product.java
+----
+
+@JmixEntity
+@Table(name = "RSTEX11_PRODUCT")
+@Entity(name = "rstex11_Product")
+public class Product {
+
+    @JoinTable(name = "RSTEX11_PRODUCT_PRODUCT_TAG_LINK",
+            joinColumns = @JoinColumn(name = "PRODUCT_ID"),
+            inverseJoinColumns = @JoinColumn(name = "PRODUCT_TAG_ID"))
+    @ManyToMany
+    private List<ProductTag> tags;
+
+    // ...
+
+}
+----
+
+In the request the `ProductTag` entity instances are references by their ids. This time, the JSON object is put into an array, as there are multiple product tags to reference.
+
+[source, http request]
+.Request with M:N references
+----
+POST http://localhost:8080/rest/entities/rstex11_Product?responseFetchPlan=product-with-tags
+
+{
+  "name": "123",
+  "price": 99.95,
+  "tags": [
+    {
+      "id": "333f3a20-c47b-4bc9-ba34-a72d2d815695" // <1>
+    },
+    {
+      "id": "c4c028f0-fec1-7512-83cd-c17537d1f502"
+    }
+  ]
+}
+----
+
+[source, json]
+.Response: 201 - Created
+----
+{
+  "_entityName": "rstex11_Product",
+  "_instanceName": "123",
+  "id": "f0e04748-dcdf-d856-2482-2904f2126fcc",
+  "createdDate": "2021-03-02",
+  "createdBy": "admin",
+  "lastModifiedDate": "2021-03-02",
+  "price": 99.95,
+  "name": "123",
+  "version": 1,
+  "tags": [
+    {
+      "_entityName": "rstex11_ProductTag",
+      "_instanceName": "shiny",
+      "id": "333f3a20-c47b-4bc9-ba34-a72d2d815695", // <2>
+      "name": "shiny"
+    },
+    {
+      "_entityName": "rstex11_ProductTag",
+      "_instanceName": "great",
+      "id": "c4c028f0-fec1-7512-83cd-c17537d1f502",
+      "name": "great"
+    }
+  ]
+}
+----
+<1> the tags are referenced as a list of JSON objects containing the ID of the already existing `ProductTag`
+<2> the response contains the stored association to the two `ProductTag` entities
+
+=== Composition Attributes
+
+For attributes that are marked as `@Composition` the situation is slightly different. As this relationship type indicates that the child entities only exists as part of the parent entity, it is also possible to directly create child entites as part of the request to create a parent.
+
+In the next example, the `OrderLine` entity is a child entity of the `Order` entity. This is expressed through the `@Composition` annotation on the `lines` attribute of the `Order` entity.
+
+[source, java]
+.Order.java
+----
+public class Order {
+    @JmixGeneratedValue
+    @Column(name = "ID", nullable = false)
+    @Id
+    private UUID id;
+
+    @Composition
+    @OneToMany(mappedBy = "order")
+    private List<OrderLine> lines;
+
+    // ...
+}
+----
+
+When an order is created through the API it is possible to directly create its order lines as part of the request. In this case all the attributes of the child entity need to be provided, in order to create it. The relationship from the parent to the child does not need to be additionally referenced. Putting the child entity into the JSON array is enough to establish the connection.
```

Comment:
Needs some punctuation.
"... created through the API, it is possible ... "
"In this case, all ..."